src/Parser/RegexMatch.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the CFONB Parser package.
+ *
+ * (c) Guillaume Sainthillier <hello@silarhi.fr>
+ * (c) @fezfez <demonchaux.stephane@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Silarhi\Cfonb\Parser;
+
+use function array_key_exists;
+use function is_string;
+use RuntimeException;
+use function strlen;
+
+class RegexMatch
+{
+    /** @var array<string, string|null> */
+    private $values;
+
+    /**
+     * @param array<string, RegexParts> $regexParts
+     * @param array<array-key, string>  $matches
+     */
+    public function __construct(array $regexParts, array $matches)
+    {
+        $values = [];
+
+        foreach (array_keys($regexParts) as $i => $key) {
+            $buildKey = $i + 1;
+            if (!array_key_exists($buildKey, $matches)) {
+                throw new RuntimeException();
+            }
+
+            $value = trim($matches[$buildKey]);
+            $values[$key] = 0 != strlen($value) ? $value : null;
+        }
+
+        $this->values = $values;
+    }
+
+    public function getString(string $key): string
+    {
+        if (!array_key_exists($key, $this->values) || !is_string($this->values[$key])) {
+            throw new RuntimeException(sprintf('key "%s" is not a string', $key));
+        }
+
+        return $this->values[$key];
+    }
+
+    public function getInt(string $key): int
+    {
+        if (!array_key_exists($key, $this->values) || !is_numeric($this->values[$key])) {
+            throw new RuntimeException(sprintf('key "%s" is not numeric', $key));
```

Comment:
Same here for `array_key_exists `exception message

Reply:
done